apps/omg/test/omg/state/core_test.exs

```diff
@@ -911,6 +913,227 @@ defmodule OMG.State.CoreTest do
            )
   end
 
+  describe "Automatic fees claiming" do
+    setup do
+      fee_claimer = OMG.TestHelper.generate_entity()
+
+      {:ok, child_block_interval} = OMG.Eth.RootChain.get_child_block_interval()
+      {:ok, state_empty} = Core.extract_initial_state(0, child_block_interval, fee_claimer.addr)
+
+      alice = OMG.TestHelper.generate_entity()
+      fees = %{@eth => %{amount: 2}}
+
+      state =
+        state_empty
+        |> do_deposit(alice, %{amount: 10, currency: @eth, blknum: 1})
+        |> Core.exec(create_recovered([{1, 0, 0, alice}], @eth, [{alice, 8}]), fees)
+        |> success?()
+
+      {:ok, [state: state, alice: alice, fees: fees, fee_claimer: fee_claimer, state_empty: state_empty]}
+    end
+
+    test "should append fee txs in block", %{state: state} do
+      state = Core.claim_fees(state)
+      {:ok, {block, _dbupdates}, _state} = form_block_check(state)
+
+      assert [payment_txbytes, fee_txbytes] = block.transactions
+
+      %Transaction.Recovered{signed_tx: %Transaction.Signed{raw_tx: payment_tx}} =
+        Transaction.Recovered.recover_from!(payment_txbytes)
+
+      %Transaction.Recovered{signed_tx: %Transaction.Signed{raw_tx: fee_tx}} =
+        Transaction.Recovered.recover_from!(fee_txbytes)
+
+      assert %Transaction.Payment{} = payment_tx
+      assert %Transaction.Fee{} = fee_tx
+    end
+
+    test "fee txs are appended even when fees aren't required", %{state: state} do
+      state = Core.claim_fees(state)
+      {:ok, {block, _dbupdates}, _state} = form_block_check(state)
+
+      assert 2 = length(block.transactions)
+    end
+
+    test "should create utxos from claimed fee", %{alice: alice, state: state, fee_claimer: fee_claimer} do
+      state = Core.claim_fees(state)
```

Comment:
bare with me here... why do we need to create UTXOs from claimed fees?

Reply:
Totally fair question - it's by [design](https://github.com/omisego/elixir-omg/blob/master/docs/fee_design.md) (HINT: fee tx output == UTXO):
> To exit fees, an operator puts a special fee transaction into a plasma chain block. This allows the operator to spend the transaction fee output as a Payment transaction. By leveraging MoreVP, we only need a contract to represent the fee transaction type and can use the pre-existing exit game contract.